src/replaying.jl

```diff
@@ -61,3 +61,36 @@ function avoid_allocations(record, f, args...; kwargs...)
     ctx = new_replay_ctx(record)
     return Cassette.overdub(ctx, f, args...; kwargs...)
 end
+
+struct FrozenFunction{F}
+    f::F
+    ctx::Dict{Tuple, ReplayCtx}
+    FrozenFunction(f) = new{typeof(f)}(f, Dict{Tuple, ReplayCtx}())
+end
+
+@generated function (f::FrozenFunction)(xs...)
+    return quote
+        if haskey(f.ctx, $xs)
+            ctx = f.ctx[$xs]
```

Comment:
We shopuld be able to work this out compile time, and not involve a dictionary in the process, but I just can't see how

Reply:
lol me neither. I mean in principle we have the information at compile time tho. Probably need a big method table globally inside this module? Store the record in it when the record is generated. Then it becomes accessible inside generated.

Reply:
maybe let's consider this later in other PR, and merge this first?

Reply:
Yes, this is not blocking